index.js

```diff
@@ -8,10 +9,24 @@ const parser = require("./src/parser");
 const printDocToDebug = require("./src/doc-debug").printDocToDebug;
 
 function parse(text, opts) {
-  if (opts.parser === "flow") {
-    return parser.parseWithFlow(text, opts.filename);
+  const parseFunction = opts.parser === "flow"
+    ? parser.parseWithFlow
+    : parser.parseWithBabylon;
+
+  try {
+    return parseFunction(text);
+  } catch (error) {
+    const loc = error.loc;
+
+    if (loc) {
+      error.codeFrame = codeFrame(text, loc.line, loc.column + 1, {
+        highlightCode: true
```

Comment:
Not sure how this will affect the browser version.

Note that colors can be turned on/off via command line switches or environment variables: https://github.com/chalk/supports-color#info